setup.py

```diff
@@ -1,79 +0,0 @@
-#!/usr/bin/python3
-# Setup file for dulwich
-# Copyright (C) 2008-2022 Jelmer Vernooĳ <jelmer@jelmer.uk>
-# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
-
-import os
-import sys
-
-from setuptools import setup
-
-if sys.platform == "darwin" and os.path.exists("/usr/bin/xcodebuild"):
-    # XCode 4.0 dropped support for ppc architecture, which is hardcoded in
-    # distutils.sysconfig
-    import subprocess
-
-    p = subprocess.Popen(
-        ["/usr/bin/xcodebuild", "-version"],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        env={},
-    )
-    out, err = p.communicate()
-    for line in out.splitlines():
-        line = line.decode("utf8")
-        # Also parse only first digit, because 3.2.1 can't be parsed nicely
-        if line.startswith("Xcode") and int(line.split()[1].split(".")[0]) >= 4:
-            os.environ["ARCHFLAGS"] = ""
-
-tests_require = ["fastimport"]
-
-
-if "__pypy__" not in sys.modules and sys.platform != "win32":
-    tests_require.extend(["gevent", "geventhttpclient", "setuptools>=17.1"])
-
-
-optional = os.environ.get("CIBUILDWHEEL", "0") != "1"
-
-# Ideally, setuptools would just provide a way to do this
-if "PURE" in os.environ or "--pure" in sys.argv:
```

Comment:
Here as well, please preserve support for the --puro flag and PURE environment variable.